lib/functions_framework/server.rb

```diff
@@ -408,30 +408,39 @@ def initialize function, config
 
       def call env
         return notfound_response if blacklisted_path? env
-        logger = env["rack.logger"] = @config.logger
-        event =
+        env["rack.logger"] = @config.logger
+        events =
           begin
             CloudEvents.decode_rack_env(env) ||
               LegacyEvents.decode_rack_env(env) ||
               raise("Unknown event type")
           rescue ::StandardError => e
             e
           end
-        response =
-          if event.is_a? CloudEvents::Event
-            logger.info "FunctionsFramework: Handling CloudEvent"
-            begin
+        response = generate_response events
+        interpret_response response
+      end
+
+      private
+
+      def generate_response events
+        logger = @config.logger
+
+        if events.is_a? Array
+          begin
+            events.each do |event|
+              logger.info "FunctionsFramework: Handling CloudEvent"
               @function.call event
-              "ok"
-            rescue ::StandardError => e
-              logger.warn e
-              e
             end
-          else
-            logger.warn e.inspect
-            string_response usage_message(e), "text/plain", 400
+            "ok"
+          rescue ::StandardError => e
+            logger.warn e
+            e
           end
-        interpret_response response
+        else
+          logger.warn e.inspect
+          string_response usage_message(e), "text/plain", 400
```

Comment:
This was broken before your change, but can you fix it while you're here? `e` isn't defined in these two lines. (Should be `events`.)